datafusion/functions/src/unicode/initcap.rs

```diff
@@ -74,7 +76,7 @@ impl ScalarUDFImpl for InitcapFunc {
             DataType::LargeUtf8 => make_scalar_function(initcap::<i64>, vec![])(args),
             DataType::Utf8View => make_scalar_function(initcap_utf8view, vec![])(args),
             other => {
-                exec_err!("Unsupported data type {other:?} for function initcap")
+                exec_err!("Unsupported data type {other:?} for function `initcap`")
```

Comment:
```suggestion
                internal_err!("Unsupported data type {other:?} for function `initcap`")
```
Incompatible arg should be checked during planning before, thus here is unreachable, and we can use `internal_err` to indicate a potential bug if it's executed

Reply:
@2010YOUY01 Thanks for reviewing,

> Incompatible arg should be checked during planning before, thus here is unreachable, and we can use internal_err to indicate a potential bug if it's executed

Make sense to me👍